highway.tex

```diff
@@ -24,807 +26,1323 @@
+$$
+   \final(B, \sigma, t) \defeq \begin{cases}
+    1 &\mbox{if } \text{there exists $(q,k)$-summit for $B$ in $\sigma$ s.t. }(2q-n)(1-2^{-k})>t\\
+    0 &\mbox{otherwise}.
+    \end{cases}
+$$
 
+Now, based on the above definition we will refer to validator $V$ as having block $B$ \emph{finalized for confidence threshold} $t$ if $t$ is an integer value s.t. $\final(B,\sigma,t) = 1$, where $\sigma$ is $V$'s protocol state.
+Next we prove the key technical lemma, showing that after a summit of certain height and quorum size occurs, it is very unlikely for new units to vote against the vote of the summit. 
 
-\subsection{Computational Efficacy of the Finality Criterion}
+\begin{lemma}\label{LemmaSummit}
+Let $C = (C_0,\dots,C_k)$ be a $(q,k)$-summit for block $B$, relative to protocol state $\bar{D}(C_0)$, $\sigma\supseteq\bar{D}(C_0)$ be any correct protocol state, and let $u$ be a unit in $\sigma$ such that $D(u)\cap C_k\neq\emptyset$ and $\vote(u)\not\geq B$.
+Then $f\geq (2q-n)(1-2^{-k})$.
+\end{lemma}
 
-\begin{theorem}
-  Given a protocol state $\sigma$, properties $p$ and $p'$ and a quorum size $q > 0$, there exists a maximal valid summit $((C_i), \sigma, p, p', q)$ in the sense that for every valid summit $((\tilde{C}_i)_i, \sigma, p, p', q)$, $\tilde{C}_i \subseteq C_i$ for all $i$.
 
-  Furthermore, there exists a polynomial time algorithm to compute $(C_i)$.
-\end{theorem}
 
 \begin{proof}
-  We show that for each $k \geq 0$, there is a unique, easily computable valid summit
-  $$((C_0, \ldots, C_k, \emptyset, \emptyset, \ldots), \sigma, p, p', q)$$
-  so that for any valid summit $((\tilde{C}_i)_{i \in \mathbb{N}}, \sigma, p, p', q)$, we have $\tilde{C}_i \subseteq C_i$ for all $i \leq k$, and that every $p'$-message that has level at least $i$ in the latter, also has level at least $i$ in the former.
 
-  The decreasing sequence $C_0 \supseteq C_1 \supseteq \ldots$ of finite sets must eventually be constant, and since each requirement in the definition of a valid summit affects only finitely many elements of it, this proves that $((C_i), \sigma, p, p', q)$ is also a valid summit and a maximal one.
+Let us define 
 
-  For $k = 0$, we let $C_0$ be the set of all validators $v \notin E(\sigma)$ whose latest $p'$-message has property $p$. Since every validator in $\tilde{C}_0$ must have at least one level $\geq 0$ message, it is clear that $\tilde{C}_0 \subseteq C_0$.
+$$A(C_0,u) = E(u)\cup \Big(S\big(\{v\in L(u) \mid \vote(v) \not\geq B\}\big) \cap E\big(C_0 \cup D(u)\big)\Big)$$
```

Comment:
Can we make the C_0 implicit, now that it's only used within the context of this proof, i.e. call it A(u) instead of A(C_0, u)?